vskernels/types.py

```diff
@@ -37,54 +37,97 @@ class BorderHandling(CustomIntEnum):
     """Assume the image was resized with extend padding, where the outermost row was extended infinitely far."""
 
     def prepare_clip(
-        self, clip: vs.VideoNode, min_pad: int = 2, shift: tuple[TopShift, LeftShift] = (0, 0)
+        self,
+        clip: vs.VideoNode,
+        width: int,
+        height: int,
+        shift: tuple[TopShift, LeftShift],
+        kernel_radius: int,
+        **kwargs: Any,
     ) -> tuple[vs.VideoNode, tuple[TopShift, LeftShift]]:
         """
         Apply required padding and adjust shift.
 
         Args:
             clip: Input clip.
-            min_pad: Minimum padding before alignment.
+            width: Output width.
+            height: Output height.
             shift: Current (top, left) shift.
+            kernel_radius: Kernel radius.
+            **kwargs: Optional src_width/src_height.
 
         Returns:
             (padded clip, updated shift).
         """
-        pad_w, pad_h = (self.pad_amount(size, min_pad) for size in (clip.width, clip.height))
 
-        if pad_w == pad_h == 0:
-            return clip, shift
+        if self is BorderHandling.MIRROR:
+            return (clip, shift)
+
+        src_width = kwargs.get("src_width", clip.width)
+        src_height = kwargs.get("src_height", clip.height)
```

Comment:
You should probably check if the variable is not None instead. Passing `src_width=None` is valid for the resize(2) plugin.